#!/usr/bin/python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

display_enum = gen.enum([gen.const("RGB", int_t, 0, "RGB Color"),
#                         gen.const("Hue", int_t, 1, "Hue Color"),
#                         gen.const("Mask", int_t, 2, "Mask Image"),
                         gen.const("Backproject", int_t, 1, "Backproject")],
                         "Enum to set result display");
gen.add("display_mode", int_t, 0, "Display Mode", 0, 0, 1, edit_method = display_enum)

gen.add("vmin", int_t, 0, "Vmin for mask image",  10, 0, 256)
gen.add("vmax", int_t, 0, "Vmax for mask image", 256, 0, 256)
gen.add("smin", int_t, 0, "Smin for mask image",  30, 0, 256)

exit(gen.generate(PACKAGE, "camshiftdemo", "camshiftdemo"))
